dispatcher/service/pool.py

```diff
@@ -70,7 +72,15 @@ async def stop(self) -> None:
 
     def cancel(self) -> None:
         self.is_active_cancel = True  # signal for result callback
-        self.process.terminate()  # SIGTERM
+
+        # If the process has never been started or is already gone, its pid may be None
+        pid = self.process.pid
+        if pid is None:
```

Comment:
I wish I was joking, but apparently MyPY is confused by the fact that you saved `pid` to a variable. using `if self.process.pid is None:` seems to fix it.

This is ridiculous, and I want to clarify I am _only_ cool with the type checking stuff because dispatcherd is lower-level than the other Django stuff, so it becomes achievable even with this stuff.